src/app/clusters/tls-client-management-server/tls-client-management-server.cpp

```diff
@@ -0,0 +1,244 @@
+{
+    VerifyOrDie(aPath.mClusterId == TlsClientManagement::Id);
+    Status status;
+
+    switch (aPath.mAttributeId)
+    {
+    case MaxProvisioned::Id: {
+        uint8_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        status = SetMaxProvisioned(newValue);
+        return StatusIB(status).ToChipError();
+    }
+    default:
+        // Unknown attribute
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+}
+
+Status TlsClientManagementServer::SetMaxProvisioned(uint8_t maxProvisioned)
+{
+    bool enableChanged = mMaxProvisioned != maxProvisioned;
+
+    if (enableChanged)
+    {
+        mMaxProvisioned = maxProvisioned;
+
+        // Write new value to persistent storage.
+        auto endpointId            = GetEndpointId();
+        ConcreteAttributePath path = ConcreteAttributePath(endpointId, TlsClientManagement::Id, MaxProvisioned::Id);
+        GetSafeAttributePersistenceProvider()->WriteScalarValue(path, mMaxProvisioned);
+
+        // and mark as dirty
+        MatterReportingAttributeChangeCallback(path);
+    }
+
+    return Protocols::InteractionModel::Status::Success;
+}
+
+void TlsClientManagementServer::InvokeCommand(HandlerContext & ctx)
+{
+    switch (ctx.mRequestPath.mCommandId)
+    {
+    case Commands::ProvisionEndpoint::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvisionEndpoint::DecodableType>(
+            ctx, [this](HandlerContext & innerCtx, const auto & req) { HandleProvisionEndpoint(innerCtx, req); });
+        break;
+    case Commands::FindEndpoint::Id:
+        CommandHandlerInterface::HandleCommand<Commands::FindEndpoint::DecodableType>(
+            ctx, [this](HandlerContext & innerCtx, const auto & req) { HandleFindEndpoint(innerCtx, req); });
+        break;
+    case Commands::RemoveEndpoint::Id:
+        CommandHandlerInterface::HandleCommand<Commands::RemoveEndpoint::DecodableType>(
+            ctx, [this](HandlerContext & innerCtx, const auto & req) { HandleRemoveEndpoint(innerCtx, req); });
+        break;
+    }
+}
+
+
+void TlsClientManagementServer::HandleProvisionEndpoint(HandlerContext & ctx, const Commands::ProvisionEndpoint::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsClientManagement: ProvisionEndpoint");
+
+    Commands::ProvisionEndpointResponse::Type response;
+    Status status = mDelegate.ProvisionEndpoint(req, response.endpointID);
+
+    if (status == Protocols::InteractionModel::Status::Success) {
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    } else {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        }
+}
+
+void TlsClientManagementServer::HandleFindEndpoint(HandlerContext & ctx, const Commands::FindEndpoint::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "TlsClientManagement: FindEndpoint");
+
+    Commands::FindEndpointResponse::Type response;
+    Status status = mDelegate.FindProvisionedEndpointByID(req.endpointID, response.endpoint);
```

Comment:
What does the lifetime management look like for the Hostname field of that endpoint struct?